packages/payload/src/utilities/flattenAllFields.ts

```diff
@@ -0,0 +1,65 @@
+import type { Field, FlattenedField } from '../fields/config/types.js'
+
+import { tabHasName } from '../fields/config/types.js'
+
+export const flattenAllFields = ({ fields }: { fields: Field[] }): FlattenedField[] => {
+  const result: FlattenedField[] = []
+
+  for (const field of fields) {
+    switch (field.type) {
+      case 'array':
+      case 'group': {
+        result.push({ ...field, flattenedFields: flattenAllFields({ fields: field.fields }) })
+        break
+      }
+
+      case 'blocks': {
+        const blocks = []
+        for (const block of field.blocks) {
+          blocks.push({
+            ...block,
+            flattenedFields: flattenAllFields({ fields: block.fields }),
+          })
+        }
+        result.push({
+          ...field,
+          blocks,
+        })
+        break
+      }
+
+      case 'collapsible':
+      case 'row': {
+        for (const nestedField of flattenAllFields({ fields: field.fields })) {
+          result.push(nestedField)
+        }
+        break
+      }
+
+      case 'tabs': {
+        for (const tab of field.tabs) {
+          if (!tabHasName(tab)) {
+            for (const nestedField of flattenAllFields({ fields: tab.fields })) {
+              result.push(nestedField)
+            }
+          } else {
+            result.push({
+              ...tab,
+              type: 'tab',
+              flattenedFields: flattenAllFields({ fields: tab.fields }),
+            })
+          }
+        }
+        break
+      }
+
+      default: {
+        if (field.type !== 'ui') {
```

Comment:
I would have made added `case 'ui': break;` for easier readability. This is fine since it is one if statement but if you want to make it better you could.

I also would prefer to make use of the `utilities/traverseFields` as much as possible. Again not a deal breaker though.